### ISSUE TITLE
Meteor > 1.4 uses node 4.x

### ISSUE THREAD

**Issue by jackboberg** [onmodulus/docker-build-meteor#25]
We can't hardcode the node version for _all_ Meteor installs. This is probably solved in [`demeteorizer`](https://github.com/onmodulus/demeteorizer), but opening the issue here for now.

Reported: onmodulus/modulus-cli#136

**Comment by jackboberg** [comment]
For those affected by this issue who need a 'quick fix', you might try demeterorizing locally and deploying the output to a node runtime:

`$ demeteorizer --json "{ \"engines\": { \"node\": \"4.4.7\" } }"`

**Comment by theworkflow** [comment]
I was thinking of a plausible solution:
- Check the Meteor version (we already do that)
- If the version is 1.4 (unfortunately it would have to equal that exact version for now) pass --json to `demeteorizer` command

It's only a hot fix, but something to get things working for now.

**Comment by jackboberg** [comment]
We currently check the version in `node` during deploy, but we would need to do it here in `bash`, right? Or create a new node script that outputs the version of `node` to use?

**Comment by theworkflow** [comment]
Yup! We should totally utilize node more in these build scripts. It makes life easier

**Comment by jackboberg** [comment]
Working on a solution on [`demeteorizer`](https://github.com/onmodulus/demeteorizer/issues/191#issuecomment-236197670)

**Comment by baberuth22** [comment]
Any luck?  I have a project that requires deployment as soon as possible and I am waiting on this issue to deploy meteor 1.4

**Comment by theworkflow** [comment]
Hey @baberuth22 we're working on a long term solution. In the meantime, you can still deploy your Meteor app to Modulus. There just requires a few changes. Here are the steps:

1) Demeteorize your app locally. You can set the npm/node version with flags. If you don't use the flags to set the node/npm version you can update the `package.json` located in `bundle/programs/server/package.json`
2) Zip your demeteorized application and deploy to Modulus under the Node.js runtime.

**Comment by theworkflow** [comment]
@baberuth22 Are you able to provide some source code that needs Meteor to use `node@v4`? If the source is deployed to Modulus can you provide me with the project id?

**Comment by jaredloson** [comment]
I tried running demeteorizer locally with `demeteorizer --json "{ \"engines\": { \"node\": \"4.4.7\" } }"`

Deployment goes through, but in the Application Log I'm getting an error: `Can't find npm module 'meteor-deque'`. Tried updating modulus CLI based on this https://github.com/onmodulus/demeteorizer/issues/175, but still getting error after each deploy.

As well, now when I run `modulus deploy` without first demeteorizing on an older version of my app that doesn't use node 4.x I get the message "You are deploying a Meteor project to a Node.js runtime. Please select the correct runtime." I have selected Node.js runtime in the administration panel. So, I have no way to get a version of my production app back up.

**_EDIT**_
After selecting the Meteor runtime in the Administration panel, I'm now able to deploy an older version of the app, but still getting the error `Can't find npm module 'meteor-deque'.

**Comment by baberuth22** [comment]
@harlanj I've had issues demeteorizing locally and using node runtime.  And it sounds like Jared is having trouble, so I'd prefer to continue as Meteor app if possible.   

I don't know that my code needs node@v4, but I've upgraded to the latest version of meteor and it's difficult to downgrade.

My project id is 79658.  If you can do something to make it work, by all means do so.   I'd be stoked!

**Comment by theworkflow** [comment]
@jaredloson @baberuth22 try running these commands when you demeteorize:

```
$ demeteorizer --server-only --node-version 4.4.7
$ npm install --production # only if you have a package.json in the root of your project
$ mv ./node_modules ./demeteorized/bundle/programs/server # only if you have a package.json in the root of your project
$ cd .demeteorized
$ modulus deploy
```

There's a flag in `demeteorizer` to specifically set the node/npm version in the generated package.json. This will handle that. Let me know how that goes.

**Comment by baberuth22** [comment]
@harlanj Running demeteorizer and converting to a nodejs project seemed to work.

> deploy -> v4.4.7
> Uploading completed build.
> Provisioning and deploying bundle to servos...
> 
> [✓] booking2 running at booking2-79658.onmodulus.net

But going to the domain I see:

Unable to connect to any application instances.

**Comment by timmyg** [comment]
@baberuth22 did you try fully stopping and starting your project?

**Comment by theworkflow** [comment]
@baberuth22 you need to change your runtime to Node.js. I recommend creating a new project and deploying the same way I informed you, set up your environment variables and all, and if it works, move your custom domain to your new project.

**Comment by jaredloson** [comment]
@harlanj tried the above but still get the npm error about not finding "meteor-deque".

Also, just noting that I do have a package.json in my root, but the node_modules folder is also in the root, so I changed the move command to `mv node_modules .demeteorized/bundle/programs/server`. I verified that the node_modules folder had indeed been moved before deploying.

I need to get a Plan B in place for an event tonight, but I may try creating a new project later and let you know how it goes.

**Comment by baberuth22** [comment]
@harlanj @timmyg I fully stopped and started my project and changed the runtime to nodejs.  

Now I didn't create a new project as I figured it was uneccessary. Also, I didn't want to regenerate app settings and desired to keep the same URL.

I guess I could try with a new project.

**Comment by baberuth22** [comment]
@harlanj I created a new node project and have the same result.

v4.4.7
deploy -> v4.4.7
Uploading completed build.
Provisioning and deploying bundle to servos...

[✓] booking3 running at booking3-86695.onmodulus.net

Unable to connect to any application instances.

**Comment by theworkflow** [comment]
Hm... Have you tried adding that package to your package.json just to ensure that it installs?

**Comment by baberuth22** [comment]
Is there a solution to "Unable to connect to any application instances."?   How do we debug that?

**Comment by theworkflow** [comment]
Check your application logs. When `demeteorizing` can you add the `--server-only` flag? I forgot to mention that @baberuth22

**Comment by baberuth22** [comment]
@harlanj That didn't work unfortunately.  

`demeteorizer --json "{ \"engines\": { \"node\": \"4.4.7\" } }" --server-only
`

Why is this issue closed?

**Comment by gschleic** [comment]
FYI, I am having the same issue and tried all the various permutations  / suggestions above. Is there a clear cut resolution to this issue?

I've also demeteorized locally and uploaded the bundle manually.

Presently I am seeing:

```

Now using node v4.4.7
Found package.json: ./programs/server/package.json
Start script specified in package.json: node ../../main
Running command: npm start

> meteor-dev-bundle@0.0.0 start /mnt/app/programs/server
> node ../../main

/mnt/app/programs/server/node_modules/fibers/future.js:280
                        throw(ex);
                        ^

Error: /mnt/app/programs/server/npm/node_modules/meteor/npm-bcrypt/node_modules/bcrypt/build/Release/bcrypt_lib.node: invalid ELF header
    at Error (native)
    at Object.Module._extensions..node (module.js:434:18)
    at Module.load (module.js:343:32)
    at Module.Mp.load (/mnt/app/programs/server/npm/node_modules/meteor/babel-compiler/node_modules/reify/node/runtime.js:16:23)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at bindings (/mnt/app/programs/server/npm/node_modules/meteor/npm-bcrypt/node_modules/bindings/bindings.js:76:44)
    at Object.<anonymous> (/mnt/app/programs/server/npm/node_modules/meteor/npm-bcrypt/node_modules/bcrypt/bcrypt.js:3:35)
    at Module._compile (module.js:409:26)

npm ERR! Linux 3.13.0-92-generic
npm ERR! argv "/mnt/app/.modulus/nvm/versions/node/v4.4.7/bin/node" "/mnt/app/.modulus/nvm/versions/node/v4.4.7/bin/npm" "start"
npm ERR! node v4.4.7
npm ERR! npm  v3.10.5
npm ERR! code ELIFECYCLE
npm ERR! meteor-dev-bundle@0.0.0 start: `node ../../main`
npm ERR! Exit status 1
```

**Comment by gschleic** [comment]
As mentioned, I am also not having much luck trying to back off to Meteor 1.3.5.1. If other have a way to do that, I am more than willing to go that route until this Meteor 1.4 messiness gets resolved on Modulus.  This is getting pretty frustrating I must say!

**Comment by Gervwyk** [comment]
Any updates on when this might be resolved. I'm getting the same issues.

**Comment by theworkflow** [comment]
@Gervwyk The fix is on staging and going through QA. The release should be before the end of the week.

**Comment by theworkflow** [comment]
@gschleic what version of meteor are you using? I've seen that error before on multiple Meteor threads, but for alpha/beta (non stable) releases.

**Comment by baberuth22** [comment]
I need to have my project deployed for a client today.  @harlanj I just sent you an email in response to your instructions.

**Comment by baberuth22** [comment]
Why am I getting this error?

~/Meteor/booking3: demeteorizer --server-only --node-version 4.4.7 --architecture os.linux.x86_64

  error: unknown option `--node-version'

**Comment by baberuth22** [comment]
I guess I can run this command instead:

demeteorizer --server-only  --architecture os.linux.x86_64  --json "{ \"engines\": { \"node\": \"4.4.7\" } }"

Is this the same?

@harlanj Can you deploy the fix to production from staging?   I have a deadline for a client today and it is the end of week 2.

**Comment by baberuth22** [comment]
Every time I deploy I get:

Unable to connect to any application instances.

So here is my terminal log:

~/Meteor/booking3:  demeteorizer --server-only  --architecture os.linux.x86_64  --json "{ \"engines\": { \"node\": \"4.4.7\" } }"
~/Meteor/booking3: cd .demeteorized/
~/Meteor/booking3/.demeteorized: modulus deploy --include-modules
Welcome to Modulus
You are logged in as sltwtr
Please choose which project to use:
  1)  booking2  Meteor
  2)  booking3  Node.js
[?] project 2
Selecting booking3

Compressing project...
20.8 MB written
Uploading project...
Upload progress [===================] 100%
Deploying project into Node.js runtime...
Starting build.
Creating directories for build environment.
Downloading source.
Executing build.
Package found: /bundle/programs/server/package.json
Installing node 4.4.7
Installing npm 3.10.6
Installing packages from /bundle/programs/server/package.json
WARN: npm-shrinkwrap.json will override dependencies declared in package.json.

> fibers@1.0.13 install /mnt/input/bundle/programs/server/node_modules/fibers
> node build.js || nodejs build.js

`linux-x64-v8-4.5` exists; testing
Binary is fine; exiting
meteor-dev-bundle@0.0.0 /mnt/input/bundle/programs/server
├── amdefine@1.0.0 
├── ansi-regex@0.2.1 
├── ansi-styles@1.1.0 
├── asap@2.0.4 
├── escape-string-regexp@1.0.5 
├── fibers@1.0.13 
├── has-ansi@0.1.0 
├── meteor-promise@0.7.2 
├── promise@7.1.1 
├── semver@4.1.0 
├── source-map@0.1.32 
├── source-map-support@0.3.2 
├── strip-ansi@0.3.0 
├── supports-color@0.2.0 
├── type-of@2.0.1 
└── underscore@1.5.2 

npm WARN meteor-dev-bundle@0.0.0 No description
npm WARN meteor-dev-bundle@0.0.0 No repository field.
npm WARN meteor-dev-bundle@0.0.0 No license field.
v4.4.7
deploy -> v4.4.7
Uploading completed build.
Provisioning and deploying bundle to servos...

[✓] booking3 running at booking3-86695.onmodulus.net

AND HERE IS THE ERROR LOG:

npm ERR! Linux 3.13.0-73-generic
npm ERR! argv "/mnt/app/.modulus/nvm/versions/node/v4.4.7/bin/node" "/mnt/app/.modulus/nvm/versions/node/v4.4.7/bin/npm" "start"
npm ERR! node v4.4.7
npm ERR! npm  v3.10.6
npm ERR! code ELIFECYCLE
npm ERR! meteor-dev-bundle@0.0.0 start: `node ../../main`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the meteor-dev-bundle@0.0.0 start script 'node ../../main'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the meteor-dev-bundle package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ../../main
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs meteor-dev-bundle
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls meteor-dev-bundle
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /mnt/app/bundle/programs/server/npm-debug.log
Now using node v4.4.7
Found package.json: ./bundle/programs/server/package.json
Start script specified in package.json: node ../../main
Running command: npm start

> meteor-dev-bundle@0.0.0 start /mnt/app/bundle/programs/server
> node ../../main

/mnt/app/bundle/programs/server/node_modules/fibers/future.js:280
                        throw(ex);
                        ^

Error: Module did not self-register.
    at Error (native)
    at Object.Module._extensions..node (module.js:434:18)
    at Module.load (module.js:343:32)
    at Module.Mp.load (/mnt/app/bundle/programs/server/npm/node_modules/meteor/babel-compiler/node_modules/reify/node/runtime.js:16:23)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at bindings (/mnt/app/bundle/programs/server/npm/node_modules/meteor/npm-bcrypt/node_modules/bindings/bindings.js:76:44)
    at Object.<anonymous> (/mnt/app/bundle/programs/server/npm/node_modules/meteor/npm-bcrypt/node_modules/bcrypt/bcrypt.js:3:35)
    at Module._compile (module.js:409:26)

npm ERR! Linux 3.13.0-73-generic
npm ERR! argv "/mnt/app/.modulus/nvm/versions/node/v4.4.7/bin/node" "/mnt/app/.modulus/nvm/versions/node/v4.4.7/bin/npm" "start"
npm ERR! node v4.4.7
npm ERR! npm  v3.10.6
npm ERR! code ELIFECYCLE
npm ERR! meteor-dev-bundle@0.0.0 start: `node ../../main`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the meteor-dev-bundle@0.0.0 start script 'node ../../main'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the meteor-dev-bundle package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ../../main
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs meteor-dev-bundle
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls meteor-dev-bundle
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /mnt/app/bundle/programs/server/npm-debug.log
Now using node v4.4.7
Found package.json: ./bundle/programs/server/package.json
Start script specified in package.json: node ../../main
Running command: npm start

> meteor-dev-bundle@0.0.0 start /mnt/app/bundle/programs/server
> node ../../main

**Comment by gschleic** [comment]
1.4.0.1

My issues on this have been enough to get me to try out Galaxy hosting from MDG. Working there so that is my work around for the time being.

**Comment by baberuth22** [comment]
Likewise.  I got my site running on Heroku without issue using this tutorial http://justmeteor.com/blog/deploy-to-production-on-heroku/

I am still using the mongo db from modulus for now.

**Comment by talfco** [comment]
@harlanj did the issue release fix date change, "before end of week" passed ? Any new estimated date for fixing available ?

**Comment by Almaju** [comment]
It's been 10 days and this critical issue has not been fixed, I am considering switching to another service. I feel like this happened too many times recently with Modulus.

**Comment by skylerapp** [comment]
+1
When does the fix go live?

**Comment by jackboberg** [comment]
Backend services were update on Thursday, and updated CLI pushed right after. Please try using

`npm i modulus@next -g`

I apologize that no one updated this thread, hopefully we have an approach now that resolves everyones issues with Meteor@1.4.x (and node 4.x).

**Comment by hrdymchl** [comment]
@jackboberg the console is still throwing the same error after using the update.

```
server/boot.js:9:13)
    at Module._compile (module.js:456:26)

npm ERR! Linux 3.13.0-66-generic
npm ERR! argv "node" "/mnt/app/.modulus/nvm/v0.10.41/bin/npm" "start"
npm ERR! node v0.10.41
npm ERR! npm  v3.10.6
npm ERR! code ELIFECYCLE
npm ERR! meteor-dev-bundle@0.0.0 start: `node ../../main`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the meteor-dev-bundle@0.0.0 start script 'node ../../main'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the meteor-dev-bundle package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ../../main
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs meteor-dev-bundle
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls meteor-dev-bundle
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /mnt/app/bundle/programs/server/npm-debug.log
Now using node v0.10.41 (npm v3.10.6)
Found package.json: ./bundle/programs/server/package.json
Running command: npm start

> meteor-dev-bundle@0.0.0 start /mnt/app/bundle/programs/server
> node ../../main


assert.js:93
  throw new assert.AssertionError({
        ^
AssertionError: "undefined" === "function"
    at wrapPathFunction (/mnt/app/bundle/programs/server/mini-files.js:77:10)
    at Object.<anonymous> (/mnt/app/bundle/programs/server/mini-files.js:108:24)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/mnt/app/bundle/programs/server/boot.js:9:13)
    at Module._compile (module.js:456:26)

npm ERR! Linux 3.13.0-66-generic
npm ERR! argv "node" "/mnt/app/.modulus/nvm/v0.10.41/bin/npm" "start"
npm ERR! node v0.10.41
npm ERR! npm  v3.10.6
npm ERR! code ELIFECYCLE
npm ERR! meteor-dev-bundle@0.0.0 start: `node ../../main`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the meteor-dev-bundle@0.0.0 start script 'node ../../main'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the meteor-dev-bundle package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ../../main
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs meteor-dev-bundle
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls meteor-dev-bundle
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /mnt/app/bundle/programs/server/npm-debug.log
```

**Comment by nickoj81** [comment]
Also receiving the exact same error after updating as suggested.

**Comment by jackboberg** [comment]
@nickoj81 do you mean the same error that @hrdymchl has posted, or that you are receiving the same _different_ error that you were getting before updating the CLI?

We also have some people sending in issues to help@modulus.io with the same `mini-files.js` line errors. Can anyone confirm that their project which required node 4.x and did not work before, now works? If so, I'd like to close this thread and open a new issue to more directly work on the `mini-files` error.

**Comment by talfco** [comment]
I did the update as @jackboberg has proposed get now the error with the mini-file.js. So still not working.

> meteor-dev-bundle@0.0.0 start /mnt/app/bundle/programs/server
> node ../../main

assert.js:93
  throw new assert.AssertionError({
        ^
AssertionError: "undefined" === "function"
    at wrapPathFunction (/mnt/app/bundle/programs/server/mini-files.js:77:10)
    at Object.<anonymous> (/mnt/app/bundle/programs/server/mini-files.js:108:24)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/mnt/app/bundle/programs/server/boot.js:9:13)
    at Module._compile (module.js:456:26)

npm ERR! Linux 3.13.0-43-generic
npm ERR! argv "node" "/mnt/app/.modulus/nvm/v0.10.41/bin/npm" "start"
npm ERR! node v0.10.41
npm ERR! npm  v3.10.6
npm ERR! code ELIFECYCLE
npm ERR! meteor-dev-bundle@0.0.0 start: `node ../../main`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the meteor-dev-bundle@0.0.0 start script 'node ../../main'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the meteor-dev-bundle package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ../../main
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs meteor-dev-bundle
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls meteor-dev-bundle
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /mnt/app/bundle/programs/server/npm-debug.log
Now using node v0.10.41 (npm v3.10.6)
Found package.json: ./bundle/programs/server/package.json
Running command: npm start

> meteor-dev-bundle@0.0.0 start /mnt/app/bundle/programs/server
> node ../../main

assert.js:93
  throw new assert.AssertionError({
        ^
AssertionError: "undefined" === "function"
    at wrapPathFunction (/mnt/app/bundle/programs/server/mini-files.js:77:10)
    at Object.<anonymous> (/mnt/app/bundle/programs/server/mini-files.js:108:24)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/mnt/app/bundle/programs/server/boot.js:9:13)
    at Module._compile (module.js:456:26)

npm ERR! Linux 3.13.0-43-generic
npm ERR! argv "node" "/mnt/app/.modulus/nvm/v0.10.41/bin/npm" "start"
npm ERR! node v0.10.41
npm ERR! npm  v3.10.6
npm ERR! code ELIFECYCLE
npm ERR! meteor-dev-bundle@0.0.0 start: `node ../../main`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the meteor-dev-bundle@0.0.0 start script 'node ../../main'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the meteor-dev-bundle package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ../../main
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs meteor-dev-bundle
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls meteor-dev-bundle
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /mnt/app/bundle/programs/server/npm-debug.log
Now using node v0.10.41 (npm v3.10.6)
Found package.json: ./bundle/programs/server/package.json

**Comment by Gervwyk** [comment]
I'm also getting the same error as talfco, even after the update. I have moved a instance to galaxy and all worked no problems on their system.. Please keep us posted. Would like to move the instance back to modulus.

**Comment by nickoj81** [comment]
@jackboberg I mean the same error as with mini-file.js

```

assert.js:93
  throw new assert.AssertionError({
        ^
AssertionError: "undefined" === "function"
    at wrapPathFunction (/mnt/app/bundle/programs/server/mini-files.js:77:10)
    at Object.<anonymous> (/mnt/app/bundle/programs/server/mini-files.js:108:24)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/mnt/app/bundle/programs/server/boot.js:9:13)
    at Module._compile (module.js:456:26)

npm ERR! Linux 3.13.0-46-generic
npm ERR! argv "node" "/mnt/app/.modulus/nvm/v0.10.41/bin/npm" "start"
npm ERR! node v0.10.41
npm ERR! npm  v3.10.6
npm ERR! code ELIFECYCLE
npm ERR! meteor-dev-bundle@0.0.0 start: `node ../../main`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the meteor-dev-bundle@0.0.0 start script 'node ../../main'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the meteor-dev-bundle package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ../../main
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs meteor-dev-bundle
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls meteor-dev-bundle
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /mnt/app/bundle/programs/server/npm-debug.log
Now using node v0.10.41 (npm v3.10.6)
Found package.json: ./bundle/programs/server/package.json
Running command: npm start
```

**Comment by jackboberg** [comment]
I've opened a new issue specifically to deal with the [`mini-files` error](https://github.com/onmodulus/docker-build-meteor/issues/28). I'm going to close this issue, since we have solved using `node@4.x` with `Meteor@1.4.x`.

```
npm i modulus@6.3 -g # currently tagged as 'next'
cd /your/project
modulus deploy --node-version 4
```

If you are trying that and still not seeing `node@4` in your servo, or having issues specifically around node version during build, please re-open this issue..otherwise let's pick up more focused conversations in separate threads.

**Comment by fvpDev** [comment]
Above instructions yielded the following (I don't see how this issue is closed)

```
Deploying Meteor version 1.4.0.1

Compressing project...
13.9 MB written
Uploading project...
Upload progress [===================] 100%
Deploying project into Meteor runtime...
Starting build.
Creating directories for build environment.
Downloading source.
Executing build.
Now using node v4.4.7 (npm v3.10.5)
Initializing Meteor environment.
Using METEOR@1.4.0.1.
Node version 4 specified.
Demeteorizing application...

[Error] There was a problem deploying your project.
```

...and that's the whole output.

**Comment by jackboberg** [comment]
@fvpDev previously a meteor build would always use `node@0.10.x`, this issue was specifically addressing the need for some users to have `node@4.x` during build/run..that is resolved.

I am opening several issues across our build and run images to address related issues now.

**Comment by fvpDev** [comment]
I tried again after deleting `.meteor/local` folder and it seems to demeteorize now (with the dependency issue for now, of course) but I'm not sure what got me the above error. Thanks for all your help Jack

**Comment by secondspace** [comment]
any updates? i'm still having this issue.

**Comment by jackboberg** [comment]
@secondspace What version of the CLI are you using? Can you open an issue specific to the problem you are having, or post to an open one?

**Comment by secondspace** [comment]
@jackboberg 

Modulus CLI is 6.3.

I'm also using Meteor 1.4.0.1

The error I'm receiving is the same as @nickoj81 is having above. 

Thank you for promptly replying. Much appreciated.

**Comment by eliasthompson** [comment]
Chipping in here. I also tried to do a deploy last night with 1.4.x on what I believe was the latest CLI, and consistently got @nickoj81's output. We had to temp deploy to Heroku. =\

**Comment by secondspace** [comment]
@jackboberg 

Seems like it demeteorizes with the correct versions of NPM & Node, but then I get this right after:

npm WARN deprecated This version of npm lacks support for important features,
npm WARN deprecated such as scoped packages, offered by the primary npm
npm WARN deprecated registry. Consider upgrading to at least npm@2, if not the
npm WARN deprecated latest stable version. To upgrade to npm@2, run:
npm WARN deprecated 
npm WARN deprecated   npm -g install npm@latest-2
npm WARN deprecated 
npm WARN deprecated To upgrade to the latest stable version, run:
npm WARN deprecated 
npm WARN deprecated   npm -g install npm@latest
npm WARN deprecated 
npm WARN deprecated (Depending on how Node.js was installed on your system, you
npm WARN deprecated may need to prefix the preceding commands with `sudo`, or if
npm WARN deprecated on Windows, run them from an Administrator prompt.)
npm WARN deprecated 
npm WARN deprecated If you're running the version of npm bundled with
npm WARN deprecated Node.js 0.10 LTS, be aware that the next version of 0.10 LTS
npm WARN deprecated will be bundled with a version of npm@2, which has some small
npm WARN deprecated backwards-incompatible changes made to `npm run-script` and
npm WARN deprecated semver behavior.
Now using node v0.10.41 (npm v1.4.29)
deploy -> v0.10.41
Installing packages from /mnt/output/bundle/programs/server/package.json
WARN: npm-shrinkwrap.json will override dependencies declared in package.json.

> fibers@1.0.13 install /mnt/output/bundle/programs/server/node_modules/fibers
> node build.js || nodejs build.js

gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
make: Entering directory `/mnt/output/bundle/programs/server/node_modules/fibers/build'
 CXX(target) Release/obj.target/fibers/src/fibers.o
 CXX(target) Release/obj.target/fibers/src/coroutine.o
 CC(target) Release/obj.target/fibers/src/libcoro/coro.o
 SOLINK_MODULE(target) Release/obj.target/fibers.node
 COPY Release/fibers.node
make: Leaving directory`/mnt/output/bundle/programs/server/node_modules/fibers/build'
Installed in `/mnt/output/bundle/programs/server/node_modules/fibers/bin/linux-x64-v8-3.14/fibers.node`
meteor-dev-bundle@0.0.0 /mnt/output/bundle/programs/server
├── amdefine@1.0.0 
├── ansi-regex@0.2.1 
├── ansi-styles@1.1.0 
├── asap@2.0.4 
├── escape-string-regexp@1.0.5 
├── fibers@1.0.13 
├── has-ansi@0.1.0 
├── meteor-promise@0.7.2 
├── promise@7.1.1 
├── semver@4.1.0 
├── source-map@0.1.32 
├── source-map-support@0.3.2 
├── strip-ansi@0.3.0 
├── supports-color@0.2.0 
├── type-of@2.0.1 
└── underscore@1.5.2

npm WARN meteor-dev-bundle@0.0.0 No description
npm WARN meteor-dev-bundle@0.0.0 No repository field.
npm WARN meteor-dev-bundle@0.0.0 No license field.
Uploading completed build.
Provisioning and deploying bundle to servos...

**Comment by jackboberg** [comment]
@nickoj81's error is resolved, see https://github.com/onmodulus/docker-build-meteor/issues/28#issuecomment-240871386

I am locking this thread, please move conversation to appropriate issues. This issue was for supporting node v4, which is definitely resolved.

For every instance I have seen so far with `mini-files`, using `modulus deploy --node-version 4` fixes it. If not, comment here: onmodulus/docker-build-meteor#28

@secondspace @eliasthompson
